src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -194,27 +194,22 @@ HRESULT WinRTGetMetadataFile(
         return E_NOTIMPL;
     }
 
-    Microsoft::WRL::Wrappers::HString substring;
-    if (SUCCEEDED(WindowsSubstringWithSpecifiedLength(name, 0, 7, substring.GetAddressOf())))
+    wchar_t szFolderPath[8];
+    StringCchPrintf(szFolderPath, _countof(szFolderPath), WindowsGetStringRawBuffer(name, nullptr));
+    if (CompareStringOrdinal(szFolderPath, 8, L"Windows", 8, false) == CSTR_EQUAL)
```

Comment:
Nit: If the input strings are null-terminated (which I thought they are), it's typically more appropriate to supply -1 as both lengths.